### PR TITLE
NetBSD path

### PR DESCRIPTION
**Pull request by nrTQgc**
It is LDC based port. My pull request merged to LDC master already. 
Druntime:
I decide try to repeat LDC team forkflow and cherry-pick my commit to ldc branch. There are a couple simple conflicts in src/core/stdc/locale.d & src/core/sys/posix/signal.d, and one absent (in master) file: src/rt/sections_ldc.d 
I successfully build and run druntime unittests on NetBSD after resolving conflicts.

### CODE REVIEW

**Comment by joakim-noah** [comment]
I suggest you split this up into 2-3 separate PRs, as it'll be a bear to review a 6k-line PR.

Are you generally just copy-pasting the FreeBSD portions, or did you check if NetBSD does it differently?  For example, `core.sys.netbsd` appears to be a straight copy of `core.sys.freebsd`, did you spend any time verifying that it is the same for NetBSD?

**Comment by nrTQgc** [comment]
Yes I checked. Actually many of them is not the same!
You can compare files:

core/sys/freebsd/time.d  and core/sys/netbsd/time.d
core/sys/freebsd/sys/event.d and core/sys/netbsd/sys/event.d

Can you suggest rule for patch splitting?

**Comment by nrTQgc** [comment]
I successfully run unit test (phobos, druntime), built dub (small issue in make file),  run vibed unit tests, started several vibe.d samples. I found one trouble with web sockets (works with firefox, core dumped with chrome). I do not discover this issue yet.

**Comment by joakim-noah** [comment]
A way to split it that makes sense to me would be to submit your `core.sys.posix` and `core.sys.netbsd` changes as separate PRs, and keep the rest in this PR.

Good to hear you got dub and vibe.d running. :)

**Comment by nrTQgc** [comment]
I splited pull request as you suggest
netbsd patch: core/sys/netbsd #1493
netbsd patch: core/sys/posix #1494

**Comment by nrTQgc** [comment]
Can we start from 1493, 1494 pull requests? May by I will change src/core/stdc/math.d (I am not sure). NetBSD has stubs for several long double (real type in dlang) functions. May be I will fix at least some of them. Actually I do it for logl function already.

File: src/core/stdc/errno.d [hunk 1]
```diff
@@ -1000,6 +1000,138 @@ else version( FreeBSD )
...
+    enum ETIME           = 92              /** STREAM ioctl timeout */;
+
+    /* File system extended attribute errors */
+    enum ENOATTR         = 93              /** Attribute not found */;
+
+    /* Realtime, XSI STREAMS option errors */
+    enum EMULTIHOP       = 94              /** Multihop attempted */; 
+    enum ENOLINK         = 95              /** Link has been severed */;
+    enum EPROTO          = 96              /** Protocol error */;
+}
```
Comment (joakim-noah) [inline src/core/stdc/errno.d]: It looks like you got these comments from the official NetBSD header, please replace them with blank comments, as seen for Linux.  Comments are copyrightable, it is Walter's policy not to copy them.

File: src/core/stdc/stdio.d [hunk 2]
```diff
@@ -166,6 +170,37 @@ else version ( FreeBSD )
...
+        int _size;
+    }
+
+    union __mbstate_t // <sys/_types.h>
+    {
+        char[128]   _mbstate8;
+        long        _mbstateL;
+    }
+}
+
```
Comment (joakim-noah) [inline src/core/stdc/stdio.d]: Remove blank line.

**Comment by joakim-noah** [comment]
Simply skimmed patch to make sure it won't mess with other platforms, ie everything is encapsulated for NetBSD, LGTM from that standpoint, plus found two formatting errors that should be fixed.

**Comment by nrTQgc** [comment]
remove comments and empty line

**Comment by joakim-noah** [comment]
Thanks, while I haven't checked the NetBSD declarations and functions, I see no harm in merging, since everything is encapsulated.

**Comment by joakim-noah** [comment]
Looks like this and the other NetBSD PRs have rotted, after nobody merged them.  Can you update them so auto-tester passes again and we'll try to get it merged?

**Comment by nrTQgc** [comment]
I will try restore context for this task. It may take some time.

**Comment by joakim-noah** [comment]
Looks like you need to rebate.

**Comment by nrTQgc** [comment]
Rebase is performed. Initially I performed rebase with wrong branch.

**Comment by nrTQgc** [comment]
auto-tester passes on this pull request, but other two requests have pending items in auto-tester. I am not sure what does it means

**Comment by joakim-noah** [comment]
This NetBSD port, including #1493 and #1494, has been in limbo for a year now.  Calling @MartinNowak, @ibuclaw, @WalterBright, can we get some feedback on what else needs to be done to get this merged?

**Comment by nrTQgc** [comment]
I performed rebase

File: src/core/stdc/tgmath.d [hunk 3]
```diff
@@ -571,6 +571,555 @@ version( FreeBSD )
...
+    ///
+    alias core.stdc.complex.cprojf     cproj;
+    ///
+    alias core.stdc.complex.cprojl     cproj;
+
+//  alias core.stdc.complex.creal      creal;
+//  alias core.stdc.complex.crealf     creal;
+//  alias core.stdc.complex.creall     creal;
+}
+
```
Comment (nrTQgc) [inline src/core/stdc/tgmath.d]: fixed in netbsd_core_sys_posix

**Pull request merged**